sjasm/directives.cpp

```diff
@@ -1801,21 +1801,21 @@ void _lua_showerror() {
 	int ln;
 
 	// part from Error(...)
-	char *err = STRDUP(lua_tostring(LUA, -1));
+	char *err = STRDUP(lua_tostring(LUA, -1)), *lnp = err;
 	if (err == NULL) {
 		Error("No enough memory!", NULL, FATAL);
 	}
-	err += 18;
-	char *pos = strstr(err, ":");
-	*(pos++) = 0;
-	ln = atoi(err) + LuaLine;
+	while (*lnp != ':' || !isdigit((unsigned char) *(lnp+1))) lnp++;
```

Comment:
while this looks definitely better than old code, I think it's still a bit too "hardcoded". If you can spend some more time on it, I would probably try to do this:
- `static int extractLineFromLuaError(char* luaError) { ... }` (then you don't need "lnp" because that will be function argument)
- inside function be more robust, i.e. check for `\0` to not run out of string in case there's unexpected string format (like lua will change from "blabla:31:something" to "blabla(31): something" in future), return simply "0" line number in case the parsing failed (not important error). Maybe even check if the string ptr is not null (in case some other code calls it in future). Or maybe even better, enclose the whole "STRDUP" inside the new function, as that is needed to do "\0" for `atoi`, and that's like implementation detail to the _show_error, can be hidden deeper. Also maybe atoi can be replaced by some internal function already used to parse numbers in sjasmplus? Then you can maybe even avoid STRDUP+free completely, because my parser takes begin/end pointers, but not sure if the code would be shorter/simpler, or worse, because it would maybe require lot of setup first, or maybe those number parsers are not visible here.

lot of thoughts, but mostly just make it survive the unexpected string format (i.e. detect `\0` for each `++lnp` to not skip over string end, the rest of it is just lot of work for small benefit.